### ISSUE TITLE
`request.authorization["token"]` is not set

### ISSUE THREAD

**Issue by milen-yordanov** [pallets/werkzeug#2662]
I have a project using 
Flask==2.2.2
Flask-HTTPAuth==4.7.0

It was working fine. Today the tests started to fail without any source code change.

I traced the problem and looks like it is caused by these refactorings
https://github.com/pallets/werkzeug/pull/2619

Flask-HTTPAuth use a code like
https://github.com/miguelgrinberg/Flask-HTTPAuth/blob/main/src/flask_httpauth.py#L395
`token = auth['token']`

But now this returns None.
Now Authorization.from_headers() returns an object that has `auth.token` but does not work as `auth['token']`

Also see: https://github.com/miguelgrinberg/Flask-HTTPAuth/issues/160

- Python version: 3.9
- Werkzeug version: 2.3.0

**Comment by davidism** [comment]
Can you please create a minimal reproducible example showing the issue using only Werkzeug? It's hard to follow the explanation you wrote.

**Comment by davidism** [comment]
OK, I see what's going on. Previously, we didn't support anything except basic and digest auth, and Flask-HTTPAuth worked around that by using a `token` key. However, now we actually support schemes with tokens, as `auth.token`. Looks like there's already a PR to address that, although they need to figure out compatibility: https://github.com/miguelgrinberg/Flask-HTTPAuth/pull/159
